indra/newview/llviewerwindowlistener.cpp

```diff
@@ -117,7 +117,10 @@ void LLViewerWindowListener::saveSnapshot(const LLSD& event) const
     {
         return response.error(stringize("Unrecognized format. [\"png\"], [\"jpeg\"] or [\"bmp\"] is expected."));
     }
-    response["result"] = mViewerWindow->saveSnapshot(filename, width, height, showui, showhud, rebuild, type, format);
+    // take snapshot on the main coro
+    doOnIdleOneTime([this, filename, width, height, showui, showhud, rebuild, type, format]()
+                    { mViewerWindow->saveSnapshot(filename, width, height, showui, showhud, rebuild, type, format); });
```

Comment:
This lambda _could_ produce a response to the invoker, but of course you'd have to change the inline error reporting from `Response` to `sendReply()`.

Reply:
Hm, I think it's worth to use `sendReply()` in this case, thanks.